src/backend/serial/scalar_mul/variable_base.rs

```diff
@@ -25,6 +25,7 @@ pub(crate) fn mul(point: &EdwardsPoint, scalar: &Scalar) -> EdwardsPoint {
     // We sum right-to-left.
 
     // Unwrap first loop iteration to save computing 16*identity
+    #[allow(unused_assignments)] // Silence warning since rustc can't tell if tmp2 is used
     let mut tmp2 = ProjectivePoint::identity();
```

Comment:
I believe this should eliminate the warning:

```suggestion
    let mut tmp2;
```